lcls_tools/superconducting/scLinac.py

```diff
@@ -545,4 +555,51 @@ def make_lcls_cryomodules(cryomoduleClass: Type[Cryomodule] = Cryomodule,
     return cryomoduleObjects
 
 
-CRYOMODULE_OBJECTS = make_lcls_cryomodules()
+linacs = {"L0B": Linac("L0B", beamlineVacuumInfixes=BEAMLINEVACUUM_INFIXES[0],
+                       insulatingVacuumCryomodules=INSULATINGVACUUM_CRYOMODULES[0]),
+          "L1B": Linac("L1B", beamlineVacuumInfixes=BEAMLINEVACUUM_INFIXES[1],
+                       insulatingVacuumCryomodules=INSULATINGVACUUM_CRYOMODULES[1]),
+          "L2B": Linac("L2B", beamlineVacuumInfixes=BEAMLINEVACUUM_INFIXES[2],
+                       insulatingVacuumCryomodules=INSULATINGVACUUM_CRYOMODULES[2]),
+          "L3B": Linac("L3B", beamlineVacuumInfixes=BEAMLINEVACUUM_INFIXES[3],
+                       insulatingVacuumCryomodules=INSULATINGVACUUM_CRYOMODULES[3])}
+
+ALL_CRYOMODULES = L0B + L1B + L1BHL + L2B + L3B
+
+
+class CryoDict(dict):
+    def __init__(self, cryomoduleClass: Type[Cryomodule] = Cryomodule,
+                 cavityClass: Type[Cavity] = Cavity,
+                 magnetClass: Type[Magnet] = Magnet, rackClass: Type[Rack] = Rack,
+                 stepperClass: Type[StepperTuner] = StepperTuner):
+        super().__init__()
+
+        self.cryomoduleClass = cryomoduleClass
+        self.cavityClass = cavityClass
+        self.magnetClass = magnetClass
+        self.rackClass = rackClass
+        self.stepperClass = stepperClass
+
+    def __missing__(self, key):
```

Comment:
It's a performance enhancement from Matt! This way you load the objects as needed instead of all at once upfront. It's pretty much a custom default dict